src/components/AssetInput/AssetInput.tsx

```diff
@@ -78,8 +81,23 @@ export const AssetInput: FC<AssetInputProps> = ({
     onLibraryClick,
     onUploadClick,
     onMultiAssetClick,
+    acceptFileType,
 }) => {
     const assetsLength = assets.length;
+    const id = useMemoizedId();
+    const inputRef = useRef<HTMLInputElement | null>(null);
+
+    const onFileChange = (event: Event) => {
+        const files = (event.target as HTMLInputElement).files;
+        if (files && onUploadClick) {
+            onUploadClick(files);
+        }
+    };
+
+    useEffect(() => {
+        inputRef?.current?.addEventListener("change", (event: Event) => onFileChange(event));
+        return inputRef?.current?.removeEventListener("change", (event: Event) => onFileChange(event));
+    }, [inputRef]);
```

Comment:
Can we maybe inline the `change` event on the `input`?
`<input onChange={...}`

Reply:
done